### PR TITLE
Parse JSON object in Optimize.

### PR DESCRIPTION
**Pull request by tofumatt**
## Summary

Addresses issue #1605

## Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/modules/optimize/setup.js [hunk 1]
```diff
@@ -240,16 +240,20 @@ class OptimizeSetup extends Component {
...
+		let { ampExperimentJSON } = this.state;
 
 		const { ampEnabled } = global.googlesitekit.admin;
 
 		if ( ! analyticsUseSnippet || ! ampEnabled ) {
 			return null;
 		}
 
+		if ( typeof ampExperimentJSON !== 'string' ) {
+			ampExperimentJSON = JSON.stringify( ampExperimentJSON );
```
Comment (aaemnnosttv) [inline assets/js/modules/optimize/setup.js:254]: Let's be sure to stringify it in a "pretty" way for display
```suggestion
			ampExperimentJSON = JSON.stringify( ampExperimentJSON, null, 2 );
```

**Comment by tofumatt** [comment]
I looked at adding a test here for this, but in the interest of time and because we'll be replacing this opted not to. Sorry about that 😓

File: assets/js/modules/optimize/setup.js [hunk 2]
```diff
@@ -151,7 +151,7 @@ class OptimizeSetup extends Component {
 				finishSetup();
 			}
 
-			getModulesData().optimize.settings.optimizeID = optimizeID;
+			getModulesData().optimize.settings = optimizeAccount;
```
Comment (aaemnnosttv) [inline assets/js/modules/optimize/setup.js:154]: This is very temporary and all but can we set the settings with the response from the server? 😄 

E.g. https://github.com/google/site-kit-wp/blob/8e56c5810cc3dc2a1ac40b8e48e27316e6e4a822/assets/js/modules/tagmanager/setup.js#L363-L369

We should be able to get the `savedSettings` from the value provided to `.then()`

Comment (tofumatt) [reply assets/js/modules/optimize/setup.js:154]: Hah, sure thing, my bad 😅

**Pull request merged**